Filter pattern docs (#4013)
twilio-labs/paste: packages/paste-website/src/pages/patterns/filter/index.mdx

import {Checkbox, CheckboxGroup} from "@twilio-paste/core/checkbox";
import {Radio, RadioGroup} from "@twilio-paste/core/radio-group";
import { Popover, PopoverButton, PopoverContainer, usePopoverState } from "@twilio-paste/core/popover";
```

## Usage

The filter pattern should be used when a user is presented with a page containing a large amount of data that they could be filtering or searching through. The filter parameters and functionality you can surface to the user will be highly dependent on the page use case.

This pattern provides recommendations for selection methods, layout, and behavior to reduce complexity and improve user efficiency when using different types of filters in a feature. Before adding filters to your feature, it's essential to:

<OrderedList>
  <ListItem>
    Familiarize yourself with your feature’s filter values and categories.
    <UnorderedList>
      <ListItem>List all filtering criteria available.</ListItem>
      <ListItem>Map out the relationships between different filters and choose the ones that will be part of your feature. Remember filters should give users a sense of control and not overwhelm them.</ListItem>
      <Blockquote

>         name="Define filter categories"

Reviewer: **todo:**  update to: Defining Helpful Filter Categories and Values for Better UX

            <Button variant="link" onClick={handleClearAll}>
              Clear all
            </Button>
          ) : null}
        </Box>
      </Box>
      <DataGrid data={filteredTableData} />
    </>
  )
}`}
/>

#### Surfacing results
<UnorderedList>
  <ListItem>When a filter value is selected by the user, the corresponding Form Pill will enter a selected state.</ListItem>
  <ListItem>When a single filter value is selected, the filter updates to display the name of the selected filter.</ListItem>
  <ListItem>When two or more filter values are selected per filter category, the total number of selected filters will be displayed in a counter badge.</ListItem>
  <ListItem>In some scenarios, it is necessary to have filter values pre-selected by default to display results. Since the selection was not made by the user, the Form Pill will remain in its default state, displaying the pre-selected filter value.</ListItem>
</UnorderedList>

 <ResponsiveImage src={FilterBarImage} alt="The filter bar has different states based on interaction" />

<DoDont>
  <Do title="Do" body="Arrange filters in a way that mirrors how users think about and interact with the data." />

>   <Dont title="Don't" body="The filter bar is designed to provide prioritized, essential filters for quick application. Avoid adding too many filters, as this can clutter the screen and make it harder for users to find relevant information." />

Reviewer: **todo:** Update this text: Don't overload the filter bar with too many filters. It's designed for prioritized, essential filters to keep the screen uncluttered and easy to navigate.

        </SidePanel>
        <SidePanelPushContentWrapper>
        // Filter components from other examples
        </SidePanelPushContentWrapper>
      </SidePanelContainer>
    </>
  )
}`}
/>

#### Surfacing results

<UnorderedList>
  <ListItem>In the “More filters” Side Panel when filters are selected, a counter Badge will be displayed in the right corner of the filter value (disclosure)</ListItem>
  <ListItem>“More filters” Button will have a counter Badge with the number of selected filter values.</ListItem>
</UnorderedList>

<DoDont>
  <Do title="Do" body="Ensure that the filters in the Filter bar and those in the Side Panel are independent and won’t need to be applied along with another filter criteria to return relevant results." />
  <Dont title="Don't" body="Don’t use a “More Filters” Side Panel for fewer than 3 additional filters." />
</DoDont>

### Add filters

An "Add filters" Popover allows users to add a filter value from a predefined list of filter options. Use it in scenarios where the list of filters may not be relevant to all users, or the user would benefit from creating their own filters set.

<Callout variant="neutral" marginY="space70">

>   <CalloutHeading as="h4">Demo is coming soon!</CalloutHeading>

Reviewer: **todo:**  Update text to: Demo coming soon!

</Callout>

> #### Surfacing results

Reviewer: **todo:**  remove the content from here until composition

              </Popover>
            </PopoverContainer>
          );
        })}
      </FormPillGroup>
      <Box display="flex" justifyContent="space-between" alignItems="center" columnGap="space30" marginTop="space50">
        <Box display="flex" columnGap="space30">
          <DetailText marginTop="space0">
            <Text as="span" color="colorTextWeak" fontSize="fontSize30">
              {filteredTableData.length} result{filteredTableData.length !== 1 && "s"}
            </Text>
          </DetailText>
          {filteredTableData.length !== data.length ? (
            <Button variant="link" onClick={handleClearAll}>
              Clear all
            </Button>
          ) : null}
        </Box>
      </Box>
      <DataGrid data={filteredTableData} />
    </>
  )
}`}
/>

Use it when filter selections naturally lead to additional criteria, or where certain filter formulas are only necessary for very specific use cases.

<DoDont>
  <Do title="Do" body="Consider the performance implications of loading conditional filters." />
  <Dont title="Don't" body="Don’t use conditional filtering for essential filter criteria." />

>   <Do title="Do" body="Only one conditional filter should be expanded at a time." />

Reviewer: **todo:** Remove this bullet

      <Heading as="h3" variant="heading40" marginBottom="space0">
        <Anchor href="/components/checkbox">Checkbox</Anchor>
      </Heading>
    </Card>
  </Column>
  <Column>
    <Card>
      <Heading as="h3" variant="heading40" marginBottom="space0">
        <Anchor href="/components/radio-group">Radio</Anchor>
      </Heading>
    </Card>
  </Column>
   <Column>
    <Card>
      <Heading as="h3" variant="heading40" marginBottom="space0">
        <Anchor href="/components/popover">Popover</Anchor>
      </Heading>
    </Card>
  </Column>
</Grid>

```jsx
// import the components for filter group patterns as required

import { FormPillGroup, useFormPillState } from "@twilio-paste/form-pill-group";
import {Input} from '@twilio-paste/input';
import {MultiselectCombobox, useMultiselectCombobox} from "@twilio-paste/combobox";
import {Button} from '@twilio-paste/button';
import {ButtonGroup} from "@twilio-paste/button-group";
import {Disclosure, DisclosureContent, DisclosureHeading} from "@twilio-paste/disclosure";
import {Badge} from "@twilio-paste/badge";
import {Checkbox, CheckboxGroup} from "@twilio-paste/checkbox";
import {Radio, RadioGroup} from "@twilio-paste/radio-group";

> import { Popover, PopoverButton, PopoverContainer, usePopoverState } from "@twilio-paste/popover";

Reviewer 2: blocking: Consumers should always import the components from core:

import { FormPillGroup, useFormPillState } from "@twilio-paste/core/form-pill-group";

```

## Usage
### General

The filter pattern should be used when a user is presented with a page containing a large amount of data that they could be filtering or searching through. The filter parameters and functionality you can surface to the user will be highly dependent on the page use case.

This pattern provides recommendations for selection methods, layout, and behavior to reduce complexity and improve user efficiency when using different types of filters in a feature. Before adding filters to your feature, it's essential to:

- Familiarize yourself with your feature’s filter values and categories.
  - List all filtering criteria available.
  - Map out the relationships between different filters and choose the ones that will be part of your feature. Remember filters should give users a sense of control and not overwhelm them.

>   - Define filter categories.
>   “A truly usable faceted search provides filter categories and filter values that are appropriate, predictable, free of jargon, and prioritized.” [Defining Helpful Filter Categories and Values for Better UX](https://www.nngroup.com/articles/filter-categories-values/).

Reviewer 3: ```suggestion
  - Define filter categories. From Nielsen Norman Group: “A truly usable faceted search provides filter categories and filter values that are appropriate, predictable, free of jargon, and prioritized” ([Defining Helpful Filter Categories and Values for Better UX](https://www.nngroup.com/articles/filter-categories-values/)).
```

  <ResponsiveImage src={UsageImage} alt="The filter category label should be the same as the form field label" />

>   - Check our [Form documentation](/components/form#selection-methods) to choose the best Selection Method for your filter use case.

Reviewer 3: ```suggestion
  - Check our [Form documentation](/components/form#selection-methods) to choose the best selection method for your filter use case.
```

- Determine users' intent and their relationship with the product and dataset.
  - Categorize filters into the most relevant and commonly used filters versus the more advanced or less used ones.

>   - Determine how many different types of users will access the feature and how much the filtering criteria will vary from use case to use case. If necessary, consider allowing users to [add custom filters](https://docs.google.com/document/d/1QcSeyk64EfHFnwVg6onVF-E-IYR1VZ91anZMBHlsNjI/edit#heading=h.h0q1xjjeczp) to provide a more tailored experience.

Reviewer 3: ```suggestion
  - Determine how many different types of users will access the feature and how much the filtering criteria will vary from use case to use case. If necessary, consider allowing users to [add custom filters](#add-filters) to provide a more tailored experience.
```

>   - Determine how many different types of users will access the feature and how much the filtering criteria will vary from use case to use case. If necessary, consider allowing users to [add custom filters](https://docs.google.com/document/d/1QcSeyk64EfHFnwVg6onVF-E-IYR1VZ91anZMBHlsNjI/edit#heading=h.h0q1xjjeczp) to provide a more tailored experience.
>   - Identify any filter values that naturally lead to additional refinement options, and consider implementing [conditional filters](https://docs.google.com/document/d/1QcSeyk64EfHFnwVg6onVF-E-IYR1VZ91anZMBHlsNjI/edit#heading=h.ml9y8ac3naoj).
> - Work with your engineer to determine any technical constraints.
>   - Ensure you have a good understanding of how the data is set up, including which filtering behaviors and combinations are not possible.
>   - Evaluate carefully the best filtering behavior for [fetching results](https://docs.google.com/document/d/1QcSeyk64EfHFnwVg6onVF-E-IYR1VZ91anZMBHlsNjI/edit#heading=h.13p2794htv04). Keep in mind user intent, but also loading times and the size of the dataset.

Reviewer 3: ```suggestion
  - Determine how many different types of users will access the feature and how much the filtering criteria will vary from use case to use case. If necessary, consider allowing users to [add custom filters](#add-filters) to provide a more tailored experience.
  - Identify any filter values that naturally lead to additional refinement options, and consider implementing [conditional filters](#conditional-filters).
- Determine any technical constraints.
  - Ensure you have a good understanding of how the data is set up, including which filtering behaviors and combinations are not possible.
  - Evaluate carefully the best filtering behavior for [fetching results](#fetching-results). Keep in mind user intent, but also loading times and the size of the dataset.
```

  - Ensure that all filter options correspond to data present in the results. Avoid including filters for data that are not visible or accessible within the dataset.
  - Evaluate the expected growth of your filtering criteria and ensure it can manage larger datasets without notable performance decline.

### Accessibility

When using the filter pattern, ensure that users:
- Can easily identify and understand labels for each selection method, filter value, or category.
- Are notified of state changes. For example, when a button changes from the disabled state to the default state, or when the dataset enters a loading state.
- Can complete all actions with the keyboard.

<Callout variant="neutral" marginY="space70">
  <CalloutText>

>     Check accessibility considerations for each component part of the filter feature, for example if you’re using Time Picker check its accessibility guidelines and make sure to follow.

Reviewer 3: ```suggestion
    Check accessibility considerations for each component part of the filter feature, for example if you’re using Time Picker, follow its accessibility guidelines.
```

  </CalloutText>
</Callout>

## Variations

### Filter bar

A filter bar is the default way of displaying filters

> Use it when all filters are relevant and necessary to the search. It can also be used in combination with a [More filters Side Panel](https://docs.google.com/document/d/1QcSeyk64EfHFnwVg6onVF-E-IYR1VZ91anZMBHlsNjI/edit#heading=h.2te8zormvjon) when there are secondary filter criteria.

Reviewer 3: ```suggestion
Use it when all filters are relevant and necessary to the search. It can also be used in combination with a [More filters Side Panel](#more-filters-side-panel) when there are secondary filter criteria.
```

> A filter bar is the default way of displaying filters
> 
> Use it when all filters are relevant and necessary to the search. It can also be used in combination with a [More filters Side Panel](https://docs.google.com/document/d/1QcSeyk64EfHFnwVg6onVF-E-IYR1VZ91anZMBHlsNjI/edit#heading=h.2te8zormvjon) when there are secondary filter criteria.

Reviewer 3: ```suggestion
A filter bar is the default way of displaying filters.

Use it when all filters are relevant and necessary to the search. It can also be used in combination with a [More filters Side Panel](#more-filters-side-panel) when there are secondary filter criteria.
```

#### Ingredients
- Heading
- Form Pill Group
- Detail Text (for number of results)
- Link-style Button (for clearing results)
- Popover
- Form and other form field components
- Primary Button (for applying filters)

> - Secondary Button (for clearing filters)

Reviewer 3: todo: add a link to each ingredient here

              </Popover>
            </PopoverContainer>
          );
        })}
      </FormPillGroup>
      <Box display="flex" justifyContent="space-between" alignItems="center" columnGap="space30" marginTop="space50">
        <Box display="flex" columnGap="space30">
          <DetailText marginTop="space0">
            <Text as="span" color="colorTextWeak" fontSize="fontSize30">
              {filteredTableData.length} result{filteredTableData.length !== 1 && "s"}
            </Text>
          </DetailText>
          {filteredTableData.length !== data.length ? (
            <Button variant="link" onClick={handleClearAll}>
              Clear all
            </Button>
          ) : null}
        </Box>
      </Box>
      <DataGrid data={filteredTableData} />
    </>
  )
}`}
/>

#### Surfacing results
- When a filter value is selected by the user, the corresponding Form Pill will enter a selected state.
- When a single filter value is selected, the filter updates to display the name of the selected filter.
- When two or more filter values are selected per filter category, the total number of selected filters will be displayed in a counter badge.

> - In some scenarios, it is necessary to have filter values pre-selected by default to display results. Since the selection was not made by the user, the Form Pill will remain in its default state, displaying the pre-selected filter value.

Reviewer 3: question: Do you know why this bulleted list has extra vertical spacing compared to the other ones?

 <ResponsiveImage src={FilterBarImage} alt="The filter bar has different states based on interaction" />

<DoDont>
  <Do title="Do" body="Arrange filters in a way that mirrors how users think about and interact with the data." />
  <Do title="Do" body="Arrange filter values in a logical order; for example, dates should be in chronological order and names should be in alphabetical order." />
  <Dont title="Don't" body="The filter bar is designed to provide prioritized, essential filters for quick application. Avoid adding too many filters, as this can clutter the screen and make it harder for users to find relevant information." />
</DoDont>

### More filters Side Panel

> Use it in scenarios where the list of filters is extensive and there are filters that are not a priority to the search. Prioritize the most relevant and commonly used filters in the filter bar for visibility, while offering more/advanced filtering options in a Side Panel.

Reviewer 3: ```suggestion
Use it in scenarios where the list of filters is extensive and there are filters that are not a priority to the search. Prioritize the most relevant and commonly used filters in the filter bar for visibility, while offering advanced filtering options in a Side Panel.
```

#### Ingredients

> - Secondary rounded Button (for more filters)
> - Side Panel
> - Neutral Counter Badge
> - Dismissible Form Pill Group (for showing filter selections)
> - Disclosure (for filter categories)
> - Link-style Button (for clearing results)

Reviewer 3: ```suggestion
- Secondary rounded Button (for more filters)
- Side Panel
- Neutral counter Badge
- Dismissible Form Pill Group (for showing filter selections)
- Disclosure (for filter categories)
- Link-style Button (for clearing results)
```

todo: also add links to these components

                      items={filter.items}
                      setSelectedCount={setSelectedCount}
                      setSelectedMoreFilters={setSelectedMoreFilters}
                      selectedMoreFilters={tempSelectedMoreFilters}
                    />
                  </DisclosureContent>
                </Disclosure>
                );
              })}
            </Box>
          </SidePanelBody>
          <SidePanelFooter>
            <ButtonGroup>
              <Button
                variant="primary"
                onClick={() => {
                  // Apply filters
                }}
              >
                Apply
              </Button>
              <Button
                variant="secondary"
                onClick={() => {
                  // Clear all filters
                }}
              >
                Clear all
              </Button>
            </ButtonGroup>
          </SidePanelFooter>
        </SidePanel>
        <SidePanelPushContentWrapper>
        // Filter components from other examples
        </SidePanelPushContentWrapper>
      </SidePanelContainer>
    </>
  )
}`}
/>

#### Surfacing results

> - In the “More filters” Side Panel when filters are selected a counter Badge will be displayed in the right corner of the filter value (disclosure).

Reviewer 3: ```suggestion
- In the “More filters” Side Panel when filters are selected, a counter Badge will be displayed in the right corner of the filter value (disclosure).
```

- “More filters” Button will have a counter Badge with the number of selected filter values.

<DoDont>
  <Do title="Do" body="Ensure that the filters in the Filter bar and those in the Side Panel are independent and won’t need to be applied along with another filter criteria to return relevant results." />

>   <Dont title="Don't" body="Don’t use a “More Filters” Side Panel for less than three additional filters." />

Reviewer 3: ```suggestion
  <Dont title="Don't" body="Don’t use a “More Filters” Side Panel for fewer than 3 additional filters." />
```

</DoDont>

### Add filters

> Add filters allow users to add a filter value from a predefined list of filter options. Use it in scenarios where the list of filters may not be relevant to all personas or the user would benefit from creating their own filters set.

Reviewer 3: ```suggestion
An "Add filters" Popover allows users to add a filter value from a predefined list of filter options. Use it in scenarios where the list of filters may not be relevant to all users, or the user would benefit from creating their own filters set.
```

<Callout variant="neutral" marginY="space70">
  <CalloutHeading as="h4">Demo is coming soon!</CalloutHeading>
</Callout>

#### Surfacing results

- Selected filters will be added to the filter bar where they will become functional.
- Added filters behave as a normal filter with the only difference being they can be removed
  - Added filters can be removed through the “Add filters” Popover or through the specific filter’s Popover.

<DoDont>
  <Do title="Do" body="Limit the primary display to max 3 prioritized filters, and offer the option to add custom filters. This behavior is designed to provide users with more choice and not overwhelm them with excessive always visible options." />

>   <Dont title="Don't" body="When your use case includes a &quot;more filters&quot; panel, avoid prompting users to add more filters, as the existing filter selection should already meet their filtering needs." />

Reviewer 3: ```suggestion
  <Dont title="Don't" body="When your use case includes a &quot;More filters&quot; panel, avoid prompting users to add more filters, as the existing filter selection should already meet their filtering needs." />
```

              </PopoverButton>
              <Popover aria-label={pill} width="size40">
                <FilterComponent value={value} onApply={onApply} popover={popover} onRemove={onRemove} /> // Create different filter components
              </Popover>
            </PopoverContainer>
          );
        })}
      </FormPillGroup>
      <Box display="flex" justifyContent="space-between" alignItems="center" columnGap="space30" marginTop="space50">
        <Box display="flex" columnGap="space30">
          <DetailText marginTop="space0">
            <Text as="span" color="colorTextWeak" fontSize="fontSize30">
              {filteredTableData.length} result{filteredTableData.length !== 1 && "s"}
            </Text>
          </DetailText>
          {filteredTableData.length !== data.length ? (
            <Button variant="link" onClick={handleClearAll}>
              Clear all
            </Button>
          ) : null}
        </Box>
      </Box>
      <DataGrid data={filteredTableData} />
    </>
  )
}`}
/>

Use it when filter selections naturally lead to additional criteria, or where certain filter formulas are only necessary for very specific use cases.

<DoDont>

>   <Do title="Do" body="Consider the performance implications of loading conditional filters." />

Reviewer 3: question for @roxanagomez : What are the performance implications of loading conditional filters?

Reviewer: Like if they're wanting to load a whole new set of filters it might take a long time to render the new components, which can affect performance.

>   <Do title="Do" body="Only one conditional filter should be expanded at a time." />

Reviewer 3: question for @roxanagomez : What does this mean? How do we prevent users from triggering multiple conditional filters?

Reviewer: mm I was thinking kind of like an open/close behavior, but it makes no sense now that I think about it. I think we can add something like:

Do: Limit conditional filters to a maximum of one per filter. 

I had something like that originally but then thought it was to restrictive, although I'm ok to just not say anything and trust designers will not add conditional filters to oblivion (?)

  <Dont title="Don't" body="Don’t use conditional filtering for essential filter criteria." />
</DoDont>

## Behavior

### Fetching results

#### Batch filtering

> Requires the user to click an “apply” button to see the results.

Reviewer 3: ```suggestion
Batch filtering requires the user to click an “Apply” button to see the results.
```

 <ResponsiveImage src={FetchingImage} alt="The filter bar supports batch filtering of results on click of apply" />

> When to use:
> - Use when there are categories with multiple interdependent filter values, and the user might want to take more time selecting the right group of filters.
> - Use when the dataset cannot be refreshed automatically and the system needs time to load the data.
> - Use when you want to prevent 'no results' scenarios.
> 
> Loading Results:
> 
> Once the user clicks the Apply button, the popover will automatically close, and a loading screen will be displayed until the data is fully loaded. It is recommended to use the [Skeleton Loader](/components/skeleton-loader) when loading the results of the filters and/or search.

Reviewer 3: ```suggestion
Use batch filtering when:
- There are categories with multiple interdependent filter values, and the user might want to take more time selecting the right group of filters.
- The dataset cannot be refreshed automatically and the system needs time to load the data.
- You want to prevent "no results" scenarios.

To load results: Once the user clicks the "Apply" button, the Popover will automatically close, and a loading screen will be displayed until the data is fully loaded. It is recommended to use the [Skeleton Loader](/components/skeleton-loader) when loading the results of the filters and/or search.
```

#### Dynamic filtering

> Filters are applied as soon as a filter selection is made.

Reviewer 3: ```suggestion
Dynamic filters are applied as soon as a filter selection is made.
```

 <ResponsiveImage src={DynamicImage} alt="The filter bar also supports batch filtering of results as soon as selection is made" />

> 
> When to use:
> - Use when the filter experience is more explorative and users need to play around with the filters to find their desired results.
> - Use when the user is expected to make quick and several filter changes during the task.
> 
> <Callout variant="neutral" marginY="space70">
>   <CalloutText>
>     Be cautious when dynamically updating results, as they can divert user attention.
>   </CalloutText>
> </Callout>
> 
> Loading Results:
> 
> Dynamic filtering is designed to allow users to experiment with different filters. The Popover will remain open until the user closes it, while the results load in the background.

Reviewer 3: ```suggestion

Use dynamic filtering when:
- The filter experience is more explorative and users need to play around with the filters to find their desired results.
- The user is expected to make multiple quick filter changes during the task.

<Callout variant="neutral" marginY="space70">
  <CalloutText>
    Be cautious when dynamically updating results, as they can divert user attention.
  </CalloutText>
</Callout>

To load results: Dynamic filtering is designed to allow users to experiment with different filters. The Popover will remain open until the user closes it, while the results load in the background.
```

> ### Empty States
> 
> When the applied filters and/or search does not return any results, use the [Empty State pattern](/patterns/empty-state) to inform the user and provide a method to reset all filter and search criteria.

Reviewer 3: ```suggestion
### Empty states

When the applied filters and/or search does not return any results, use the [empty state pattern](/patterns/empty-state) to inform the user and provide a method to reset all filter and search criteria.
```

                  <FilterPillView label={filterMap[pill].label} selectedType={isSelected ? pill : null} selectedValue={value} />
                </FormPill>
              </PopoverButton>
              <Popover aria-label={pill} width="size40">
                <FilterComponent value={value} onApply={onApply} popover={popover} onRemove={onRemove} /> // Create different filter components
              </Popover>
            </PopoverContainer>
          );
        })}
      </FormPillGroup>
      <Box display="flex" justifyContent="space-between" alignItems="center" columnGap="space30" marginTop="space50">
        <Box display="flex" columnGap="space30">
          <DetailText marginTop="space0">
            <Text as="span" color="colorTextWeak" fontSize="fontSize30">
              {filteredTableData.length} result{filteredTableData.length !== 1 && "s"}
            </Text>
          </DetailText>
          {filteredTableData.length !== data.length ? (
            <Button variant="link" onClick={handleClearAll}>
              Clear all
            </Button>
          ) : null}
        </Box>
      </Box>
      <DataGrid data={filteredTableData} />
    </>
  )
}`}
/>

> ### Clearing Filters

Reviewer 3: ```suggestion
### Clearing filters
```

A clear filter option will be available when filters are selected. By default, "clear all filters" will remove all filter criteria and display all results. In use cases where there is a selected filter value by default, "clear all filters" will reset the filters to their default state.

<DoDont>
  <Do title="Do" body="Provide ways to clear all filters as a global action." />
  <Do title="Do" body="Provide ways to clear all filters on a filter category level." />
</DoDont>

> ### Responsive
> 
> On mobile screens, consolidate all filter options under a "Filter" button to optimize screen space and usability.

Reviewer 3: ```suggestion
### Responsiveness

On small screen sizes, consolidate all filter options under a "Filter" button to optimize screen space and usability.
```

> ### Responsive
> 
> On mobile screens, consolidate all filter options under a "Filter" button to optimize screen space and usability.
> 
>  <ResponsiveImage src={Responsive1Image} alt="On smaller screens, filter modal will take full width" />
> 
> 
> On responsive screens, filter pills will stack vertically to make better use of available space. All action buttons (such as "More filters", “Add Filters”, "Clear All” and “table actions") to the last row on responsive screens to maintain a clean and organized layout.

Reviewer 3: ```suggestion
### Responsive

On mobile screen sizes, consolidate all filter options under a "Filter" button to optimize screen space and usability.

 <ResponsiveImage src={Responsive1Image} alt="On smaller screens, filter modal will take full width" />

At smaller screen sizes, filter Pills will overflow onto the next row. All other actions (such as "More filters", “Add filters”, "Clear all” and table actions) move to the last row to maintain a clean and organized layout.
```

 <ResponsiveImage src={Responsive2Image} alt="filter pills will stack vertically to make better use of available space" />

> ## Usage Guide
> 
> ### Figma
> 
> #### Example
> 
> Form Pill Group is available in the Paste Components Figma library. It has 1 base component, Form Pill, with a ⚙️ emoji in its layer name to indicate there are nested configurable properties available.
> 
> In code, when there are more pills than the max width, they will wrap to the next row. To accomplish the same behavior in Figma, please use the `Row count` property to select a number of rows, or place multiple Form Pill Group components under one another.
> 
> Form Pill Group also contains other Paste components as nested components, including [Avatar](/components/avatar) and [Icon](/components/icons).
> 
> 

Reviewer 3: ```suggestion
```
